README.md

```diff
@@ -72,74 +80,153 @@ const App = () => (
+
+While `<Home path="/" />` is completely equivilant to `<Route path="/" component={Home} />`, TS users may find the latter preferable. 
+
+```js
+import { LocationProvider, Router, Route } from 'preact-iso';
 
 const App = () => (
 	<LocationProvider>
-		<ErrorBoundary>
-			<Router>
-				<Home path="/" />
-				<Profiles path="/profiles" />
-				<Profile path="/profiles/:id" />
-			</Router>
-		</ErrorBoundary>
+		<Router>
+			{/* Both of these are equivalent */}
+			<Home path="/" />
+			<Route path="/" component={Home} />
+
+			<Profile path="/profile" />
+			<NotFound default />
+		</Router>
 	</LocationProvider>
 );
 ```
 
-During prerendering, the generated HTML includes our full `<Home>` and `<Profiles>` component output because it waits for the `lazy()`-wrapped `import()` to resolve.
+Props for any route component:
 
-You can use the `useRoute` hook to get information of the route you are currently on.
+ - `path: string` - The path to match (read on)
+ - `default?: boolean` - If set, this route is a fallback/default route to be used when nothing else matches
 
-**Progressive Hydration:** When the app is hydrated on the client, the route (`Home` or `Profile` in this case) suspends. This causes hydration for that part of the page to be deferred until the route's `import()` is resolved, at which point that part of the page automatically finishes hydrating.
+Specific to the `Route` component:
 
-**Seamless Routing:** Switch switching between routes on the client, the Router is aware of asynchronous dependencies in routes. Instead of clearing the current route and showing a loading spinner while waiting for the next route (or its data), the router preserves the current route in-place until the incoming route has finished loading, then they are swapped.
+ - `component: AnyComponent` - The component to render when the route matches
+
+#### Path Segment Matching
+
+Paths are matched using a simple string matching algorithm. The following features may be used:
+
+ - `:param` - Matches any URL segment (can later extract this value from `useRoute()`)
+   - `/profile/:id` will match `/profile/123` and `/profile/abc`
+   - `/profile/:id?` will match `/profile` and `/profile/123`
+ - `*` - Matches any number of URL segments
+   - `/profile/*` will match `/profile`, `/profile/123`, `/profile/123/abc`, etc.
+
+These can then be composed to create more complex routes:
+
+ - `/profile/:id/*` will match `/profile/123`, `/profile/123/abc`, `/profile/123/abc/def`, etc.
+
+
+### `useLocation`
+
+A hook to work with the `LocationProvider` to access location context.
+
+Props:
+
+ - `url: string` - *Redundant* - The current path
+ - `path: string` - The current path
+ - `query: Record<string, string>` - The current query string parameters (`/profile?name=John` -> `{ name: 'John' }`)
+ - `route: (url: string, replace?: boolean) => void` - A function to programmatically navigate to a new route
```

Comment:
The `replace` parameter might not be obvious to everybody.

```suggestion
 - `route: (url: string, replace?: boolean) => void` - A function to programmatically navigate to a new route. `replace` will overwrite the current slot in history. So hitting the back button will not show this page but the one before in history.
```

Reply:
Good call